rust/src/datafusion/logical_expr.rs

```diff
@@ -116,6 +116,24 @@ pub fn resolve_expr(expr: &Expr, schema: &Schema) -> Result<Expr> {
     }
 }
 
+/// Coerce logical expression for filters to bollean.
+///
+/// Parameters
+///
+/// - *expr*: a datafusion logical expression
+pub fn coerce_filter_type_to_boolean(expr: &Expr) -> Result<Expr> {
+    match expr {
+        // TODO: consider making this dispatch more generic, i.e. fun.output_type -> coerce
+        // instead of hardcoding coerce method for each function
+        Expr::ScalarFunction {
+            fun: BuiltinScalarFunction::RegexpMatch,
+            args: _,
+        } => Ok(Expr::IsNotNull(Box::new(expr.to_owned()))),
+
+        _ => Ok(expr.clone()),
+    }
+}
```

Comment:
If we can do this with an owned value, I think we prefer that. If needed, the caller can clone the value themselves.

```suggestion
pub fn coerce_filter_type_to_boolean(expr: Expr) -> Result<Expr> {
    match expr {
        // TODO: consider making this dispatch more generic, i.e. fun.output_type -> coerce
        // instead of hardcoding coerce method for each function
        Expr::ScalarFunction {
            fun: BuiltinScalarFunction::RegexpMatch,
            args: _,
        } => Ok(Expr::IsNotNull(Box::new(expr))),

        _ => Ok(expr),
    }
}
```